handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -303,9 +306,19 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     private void replaceHandler(ChannelHandlerContext ctx, Selection selection) {
+        SSLEngine sslEngine = null;
         this.selection = selection;
-        SslHandler sslHandler = selection.context.newHandler(ctx.alloc());
-        ctx.pipeline().replace(this, SslHandler.class.getName(), sslHandler);
+        try {
+            sslEngine = selection.context.newEngine(ctx.alloc());
+            ctx.pipeline().replace(this, SslHandler.class.getName(), selection.context.newHandler(sslEngine));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2209)